Bug/fix rcmd check

arunge opened the pull request:
# BMSC version 24.09.1

## Bugfixes
- fix issues with RCMD check (#8)

arunge commented:
@mgross Please check this comment: https://github.com/Pandora-IsoMemo/bmsc/issues/8#issuecomment-2363645457

jan-abel-inwt commented:
https://github.com/Pandora-IsoMemo/bmsc/issues/8#issuecomment-2363627061
> Can we already remove all the content of the src folder? Or is there something missing?

@arunge i removed the content of the src folder. We now only "track" the empty directory. I also updated the installation instruction in the Readme

jan-abel-inwt commented:
-  [x] add r-cmd check - see https://github.com/Pandora-IsoMemo/docs/issues/12
-  [x] add online doc - see https://github.com/Pandora-IsoMemo/docs/issues/11

arunge commented:
@jan-abel-inwt To fix issues with the checks I
- added an argument to control the number of cores and decreased the cores to two in the test
- updated the documentation and added missing entries
- updated the `buildignore` to include the `Dockerfile`

There is one issue left in the checks that gives a warning:

```
❯ checking for GNU extensions in Makefiles ... WARNING
  Found the following file(s) containing GNU extensions:
    src/Makevars
  Portable Makefiles do not use GNU extensions such as +=, :=, $(shell),
  $(wildcard), ifeq ... endif, .NOTPARALLEL See section ‘Writing portable
  packages’ in the ‘Writing R Extensions’ manual.

❯ checking installed package size ... NOTE
    installed size is 93.8Mb
    sub-directories of 1Mb or more:
      libs  93.2Mb

❯ checking compilation flags used ... NOTE
  Compilation used the following non-portable flag(s):
    ‘-Wdate-time’ ‘-Werror=format-security’ ‘-Wformat’

0 errors ✔ | 1 warning ✖ | 2 notes ✖
Error: R CMD check found WARNINGs
Execution halted
Error: Process completed with exit code 1.
```

However, since we are using `rstantools::rstan_config()`, the `Makevars` file is generated by `rstantools`. This seems to be an issue (from ChatGPT):

> The Makevars file generated by rstantools::rstan_config() uses several GNU Makefile-specific commands, particularly $(shell ...), which is responsible for executing shell commands within the Makefile. Since this is a GNU extension, it triggers the warning during R CMD check.

Do you have an idea how to solve this?

jan-abel-inwt commented:
Thank you, @arunge, for the fix. I had already prepared one in the INWT R Docker base image, but it's still under review. Your fix seems to work fine.

The error about the Compiler Flags was a bit tricky. The `rstan_config` function does not seem to set the `MakeVars` file correctly, so that `devtools::check()` runs without warnings. I created a small script that creates the MakeVars file without using shell variables. I also adjusted the workflow files for GitHub Actions. Everything seems to be fine now.

jan-abel-inwt reviewed (approved):
see note above
